auth/src/swig/auth.i

```diff
@@ -1179,6 +1182,124 @@ static CppInstanceManager<Auth> g_auth_instances;
       });
     return taskCompletionSource.Task;
   }
+
+  /// @deprecated This method is deprecated in favor of methods that return
+  /// `Task<AuthResult>`. Please use
+  /// @ref LinkWithCredentialAsync(Credential) instead.
+  ///
+  /// Links the user with the given 3rd party credentials.
+  ///
+  /// For example, a Facebook login access token, a Twitter token/token-secret
+  /// pair.
+  /// Status will be an error if the token is invalid, expired, or otherwise
+  /// not accepted by the server as well as if the given 3rd party
+  /// user id is already linked with another user account or if the current user
+  /// is already linked with another id from the same provider.
+  ///
+  /// Data from the Identity Provider used to sign-in is returned in the
+  /// @ref AdditionalUserInfo inside @ref SignInResult.
+  [System.Obsolete("Please use `Task<AuthResult> LinkWithProviderAsync(Credential)` instead", false)]
+  public System.Threading.Tasks.Task<SignInResult> LinkAndRetrieveDataWithCredentialAsync_DEPRECATED(
```

Comment:
Techically this one doesn't need a DEPRECATED suffix since there's no conflict in a new overloaded version of the method -- it's going to be simply removed in the future, not replaced now.

Reply:
In that case, C++ should be updated as well

https://github.com/firebase/firebase-cpp-sdk/pull/1290